gardener/container_network_metrics_provider_linux.go

```diff
@@ -2,90 +2,136 @@ package gardener
 
 import (
 	"bytes"
+	"flag"
 	"fmt"
-	"path/filepath"
+	"os"
 	"strconv"
 	"strings"
 
 	"code.cloudfoundry.org/garden"
+	"code.cloudfoundry.org/guardian/kawasaki/netns"
 	"code.cloudfoundry.org/lager/v3"
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/vishvananda/netlink"
 )
 
-type SysFSContainerNetworkMetricsProvider struct {
-	containerizer   Containerizer
-	propertyManager PropertyManager
+func init() {
+	reexec.Register("fetch-container-network-metrics", func() {
```

Comment:
Similar approach to what was done in https://github.com/cloudfoundry/guardian/blob/8789f016ea2d285889aa47947092b5665b24605c/kawasaki/configure/container.go#L26